feat: add conditional blocks in docs based off release type

sklppy88 opened the pull request:
This pr adds conditional blocks, and new version / network tags that get replaced based on the release type

critesjosh commented on docs/README.md:
```diff
@@ -318,6 +318,94 @@ This value may be different from the `#include_aztec_version` macro, since the t
+| nightly | `COMMIT_TAG` | `3.0.0-nightly.20251218` |
+| devnet | `DEVNET_TAG` | `3.0.0-devnet.5` |
+| testnet | `TESTNET_TAG` | `2.1.9` |
+| mainnet | `MAINNET_TAG` | `1.0.0` |
```
I think the default value is currently 2.1.9

critesjosh commented on docs/src/preprocess/include_version.js:
```diff
@@ -1,13 +1,95 @@
+function getReleaseNetwork(releaseType) {
+  switch (releaseType) {
+    case "nightly":
+      return "sandbox";
```
I think this should be "local network" or something similar. Alejo recently had a PR to change all references of "sandbox" to "local network"

critesjosh commented on docs/src/preprocess/include_version.js:
```diff
@@ -1,13 +1,95 @@
+    case "ignition":
+      // For mainnet/ignition, use mainnet tag and strip 'v' prefix
+      return mainnetTag.startsWith("v") ? mainnetTag.substring(1) : mainnetTag;
+    default:
```
the default case falls back to testnet version. should this throw an error instead since we already validate releaseType above? this code path should be unreachable.

critesjosh commented on docs/src/preprocess/conditional_content.js:
```diff
@@ -0,0 +1,204 @@
+ * Processes conditional content blocks in markdown based on RELEASE_TYPE.
+ *
+ * Syntax:
+ *   #if(condition) {
```
do we actually use this anywhere? I am reluctant to add this functionality if we aren't using it, since its just another thing that could break 😅

sklppy88 replied:
yes, as discussed in the meet friday, for a decent amount of node docs / links / instructions 🙏

critesjosh commented on docs/src/preprocess/conditional_content.js:
```diff
@@ -0,0 +1,204 @@
+  let matched = false;
+
+  // Match the initial #if(condition) { content }
+  const ifPattern = /^#if\((\w+)\)\s*\{([\s\S]*?)\}/;
```
the regex `\{([\s\S]*?)\}`uses non-greedy matching, which may not handle content containing literal } characters correctly (e.g., code blocks with JSON). consider a more robust brace-matching approach if this becomes an issue.

sklppy88 replied:
have handled by using newline delineations instead of {} based

critesjosh commented on docs/src/preprocess/include_version.js:
```diff
@@ -1,13 +1,101 @@
+  // Get environment variables
+  // NIGHTLY_TAG: version for nightly releases (e.g., "v3.0.0-nightly.20251222")
+  // COMMIT_TAG: kept for backwards compatibility with #include_aztec_version
+  const nightlyTag = process.env.NIGHTLY_TAG || process.env.COMMIT_TAG || "0.0.0-nightly.0";
```
the README and CLAUDE.md should document this updated behavior

critesjosh reviewed (commented):
what do you think about adding some unit tests to ensure this parsing is working correctly?

Also, can you have claude do a pass and update docs/CLAUDE.md with the changes as well?

sklppy88 commented:
> what do you think about adding some unit tests to ensure this parsing is working correctly?
> 
> Also, can you have claude do a pass and update docs/CLAUDE.md with the changes as well?

Good call, have done so !

sklppy88 committed:
feat: add conditional blocks in docs based off release type

This pr adds conditional blocks, and new version / network tags that get replaced based on the release type
